extensions/wikia/ArticleVideo/bower_components/html5-skin/js/controller.js

```diff
@@ -0,0 +1,1496 @@
+      }
+    },
+
+    onPaused: function(event, videoId) {
+      if (videoId != this.focusedElement || this.state.screenToShow == CONSTANTS.SCREEN.END_SCREEN) { return; }
+      if (videoId == OO.VIDEO.MAIN && this.state.screenToShow != CONSTANTS.SCREEN.AD_SCREEN && this.state.screenToShow != CONSTANTS.SCREEN.LOADING_SCREEN) {
+        if (this.state.duration - this.state.mainVideoPlayhead < 0.01) { //when video ends, we get paused event before played event
+          this.state.pauseAnimationDisabled = true;
+        }
+        if (this.state.pauseAnimationDisabled == false && this.state.discoveryData && this.skin.props.skinConfig.pauseScreen.screenToShowOnPause === "discovery"
+            && !(Utils.isIPhone() || (Utils.isIos() && this.state.fullscreen))) {
+          OO.log("Should display DISCOVERY_SCREEN on pause");
+          this.sendDiscoveryDisplayEvent("pauseScreen");
+          this.state.screenToShow = CONSTANTS.SCREEN.DISCOVERY_SCREEN;
+        } else if (this.skin.props.skinConfig.pauseScreen.screenToShowOnPause === "social") {
+          // Remove this comment once pause screen implemented
+        } else {
+          // default
+          this.state.screenToShow = CONSTANTS.SCREEN.PAUSE_SCREEN;
+        }
+        if (Utils.isIPhone()){
+          //iPhone pause screen is the same as start screen
+          this.state.screenToShow = CONSTANTS.SCREEN.PAUSE_SCREEN;
+        }
+        this.state.playerState = CONSTANTS.STATE.PAUSE;
+        this.state.mainVideoElement.addClass('oo-blur');
+        this.renderSkin();
+      }
+      else if (videoId == OO.VIDEO.ADS){
+        this.state.adPauseAnimationDisabled = false;
+        this.state.playerState = CONSTANTS.STATE.PAUSE;
+        this.renderSkin();
+      }
+      if (this.pausedCallback) {
+        this.pausedCallback();
+        this.pausedCallback = null;
+      }
+    },
+
+    onPlayed: function() {
+      var duration = this.state.mainVideoDuration;
+      this.state.duration = duration;
+      this.skin.updatePlayhead(duration, duration, duration);
+
+      if (this.state.upNextInfo.delayedSetEmbedCodeEvent) {
+        var delayedContentData = this.state.upNextInfo.delayedContentData;
+        this.state.screenToShow = CONSTANTS.SCREEN.LOADING_SCREEN;
+
+        if (delayedContentData.clickedVideo.embed_code){
+          this.mb.publish(OO.EVENTS.SET_EMBED_CODE, delayedContentData.clickedVideo.embed_code, this.state.playerParam);
+        }
+        else if (delayedContentData.clickedVideo.asset){
+          this.mb.publish(OO.EVENTS.SET_ASSET, delayedContentData.clickedVideo.asset, this.state.playerParam);
+        }
+
+        this.mb.publish(OO.EVENTS.DISCOVERY_API.SEND_CLICK_EVENT, delayedContentData);
+        this.state.upNextInfo.delayedSetEmbedCodeEvent = false;
+        this.state.upNextInfo.delayedContentData = null;
+      }
+      else if (this.state.discoveryData && this.skin.props.skinConfig.endScreen.screenToShowOnEnd === "discovery"
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Reduce the number of conditional operators (4) used in the expression (maximum allowed 3). [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3AS1067)